modules/renter/contractor/editor.go

```diff
@@ -116,6 +116,7 @@ func (he *hostEditor) Upload(data []byte) (crypto.Hash, error) {
 	he.contractor.persist.update(updateUploadRevision{
 		NewRevisionTxn:     contract.LastRevisionTxn,
 		NewSectorRoot:      sectorRoot,
+		NewSectorIndex:     len(contract.MerkleRoots) - 1,
```

Comment:
double checking, don't you want this to be len(contract.MerkleRoots) - so you append, instead of replace the most recent value?